test/pdfboxing/overlay_test.clj

```diff
@@ -0,0 +1,27 @@
+(ns pdfboxing.overlay-test
+  (:require
+    [clojure.java.io :as io]
+    [clojure.test :refer [deftest is]]
+    [pdfboxing.common :as common]
+    [pdfboxing.overlay :refer [overlay-pdf]]))
+
+(def output-file-path "test/pdfs/foo.pdf")
+
+(deftest pdf-file-merge
+  (let [_
+        (overlay-pdf
+          {:input-file-path "test/pdfs/clojure-1.pdf"
+           :overlay-file-path "test/pdfs/clojure-2.pdf"
+           :output-file-path output-file-path})
+
+        overlayed-pdf-file (.exists (io/as-file output-file-path))]
+    (is (true? overlayed-pdf-file))
+    (is (true? (common/is-pdf? output-file-path)))))
+
+;; clean up
+(defn clean-up [file]
+  (if (.exists (io/as-file file))
+    (io/delete-file file)))
+
+(deftest cleaner
+  (clean-up "test/pdfs/foo.pdf"))
```

Comment:
Some more feedback here as well!

1. Keep in mind that not all test runners are deterministic! I don't know about clj test-runner but others like kaocha allow you to randomize the test run sequence to catch side-effects. Therefore you should not make assumptions about the order tests run. Consider cleaning up the temporary file in the deftest it was created ;).
2. What if the output path already exists for some reason? Your code will overwrite that and then delete it. That shouldn't happen. As an alternative, take a look at this: https://stackoverflow.com/a/57845906

You could also consider the following function:

```
(defn with-temp-pdf [prefix fn]
  (let [file (java.io.File/createTempFile prefix ".pdf")]
    (fn (str file))
    (if (.exists file)
      (io/delete-file file))
```

Reply:
Very good points! I followed this pattern from the tests in `merge`.
I'll update once the byte array vs output file is finalised.